### PR TITLE
Helm Chart for pgo-deployer

### LINKED ISSUE #1338: Helm install
Using Helm is quite a standard nowadays and other Postgres operators can be installed using it. Could there be a chart for installation instead of a bash install and Ansible ? 

Here are some advantages :

- Fewer documentation to keep up to date
- Easier install (one command line `helm install` instead of `make setupnamespaces; make installrbac; makedeployoperator`)
- No need to use an expenv util (I noticed removal of this was done for [4.2.2 release ](https://github.com/CrunchyData/postgres-operator/releases/tag/v4.2.2))
- Easier way to expose operator parameters to end user (example [here](https://github.com/CrunchyData/postgres-operator/issues/1337))

### PR DESCRIPTION
**Pull request by jmckulk**
This change adds the a helm chart that sets up and runs the pgo-deployer job. The
chart follows Helm best practices for ths installer.

**Checklist:**

 
 - [ ] Have you added an explanation of what your changes do and why you'd like them to be included?
 - [ ] Have you updated or added documentation for the change, as applicable?
 - [ ] Have you tested your changes on all related environments with successful results, as applicable?

**Type of Changes:**

 
 - [ ] Bug fix (non-breaking change which fixes an issue)
 - [ ] New feature (non-breaking change which adds functionality)
 - [ ] Breaking change (fix or feature that would cause existing functionality to change)

**What is the current behavior? (link to any open issues here)**

**What is the new behavior (if this is a feature change)?**

**Other information**:
[ch8484]
closes #1338

### CODE REVIEW

File: docs/content/installation/other/helm.md [hunk 1]
```diff
@@ -0,0 +1,233 @@
...
+
+- Installation
+- Upgrading
+- Uninstallation
+
+After configuring the `values.yaml` file with you configuration options, the
+installer will be run using the `helm` command line tool and takes care of
+setting up all of the objects required to run the PostgreSQL Operator.
+
+The Helm chart is available in the [`installers/helm/`]() directory path in the
```
Comment (jmckulk) [inline docs/content/installation/other/helm.md:23]: needs link

File: installers/helm/.helmignore [hunk 2]
```diff
@@ -0,0 +1,23 @@
+# Patterns to ignore when building packages.
```
Comment (jmckulk) [inline installers/helm/.helmignore:1]: this file was generated by `helm create` when the chart was created. Is this something that we should keep?

File: installers/helm/README.md [hunk 3]
```diff
@@ -0,0 +1,53 @@
...
+This Helm chart uses the `pgo-deployer` container to install the Crunchy PostgreSQL
+Operator. Helm will setup the ServiceAccount, RBAC, and ConfigMap needed to
+run the container as a Kubernetes Job. Then a job will be created based on `helm`
+`install`, `upgrade`, or `uninstall`. After the job has completed the RBAC will
+be cleaned up.
+
+## Prerequisites
+
+- Helm v3.2.3
+- Kubernetes 1.14+
```
Comment (jmckulk) [inline installers/helm/README.md:12]: Should I update these prereqs? I have developed and tested these changes with Helm v3.2.3 and multiple versions of kube. The oldest version of kube that i have had access to is 1.14 in gke

File: installers/helm/README.md [hunk 4]
```diff
@@ -0,0 +1,53 @@
...
+The following shows the configurable parameters that are relevant to the Helm
+Chart. A full list of all Crunchy PostgreSQL Operator configuration options can
+be found in the [documentation](https://crunchydata.github.io/postgres-operator/latest/installation/configuration/).
+
+| Name | Default | Description |
+| ---- | ------- | ----------- |
+| fullnameOverride | "" |  |
+| rbac.create | true | If false rbac will not be created. RBAC resources will need to be created manually and bound to `serviceAccount.name` |
+| serviceAccount.create | true | If false a service account will not be created. ServiceAccount will need to be created manually |
+| serviceAccount.name | "" | Use to override the default service account name. If serviceAccount.create is false this service account will be used. |
```
Comment (jmckulk) [inline installers/helm/README.md:53]: need to add the new var `rbac.useClusterAdmin`

File: installers/helm/templates/NOTES.txt [hunk 5]
```diff
@@ -0,0 +1,23 @@
...
+        / /                             '.
+       /_|       )                     .\|
+         |      /\                     . '
+         '--.__|  '--._  ,            /
+                      /'-,          .'
+                     /   |        _.' 
+                    (____\       /    
+                          \      \    
+                           '-'-'-'
+
```
Comment (jmckulk) [inline installers/helm/templates/NOTES.txt:21]: should we add a copyright for the ascii hippo? Do we have a crunchy ascii hippo?

File: installers/helm/templates/postgres-operator-install.yaml [hunk 6]
```diff
@@ -0,0 +1,13 @@
...
+apiVersion: batch/v1
+kind: Job
+metadata:
+  name: pgo-deploy
+  namespace: {{ .Release.Namespace }}
+  labels:
+{{ include "postgres-operator.labels" . | indent 4 }}
+  annotations:
+    helm.sh/hook: post-install
+    helm.sh/hook-delete-policy: hook-succeeded,before-hook-creation
```
Comment (jmckulk) [inline installers/helm/templates/postgres-operator-install.yaml:12]: I added `before-hook-creation` here to help if the job is run but fails. In that case the resources are not cleaned up and the install cannot be run again. By using `hook-succeeded` and  `before-hook-creation` the failed job will be kept around until a new command is run allowing for the user to look at the failed logs. Would this be good to document?

File: installers/helm/README.md (hunk 3 above)
Comment (jmckulk) [reply installers/helm/README.md:12]: I tested with helm 3.0.0 and didn't run into any issues

File: installers/helm/.helmignore (hunk 2 above)
Comment (jkatz) [reply installers/helm/.helmignore:1]: No

File: installers/helm/Chart.yaml [hunk 7]
```diff
@@ -0,0 +1,12 @@
+apiVersion: v2
+name: postgres-operator
+description: Crunchy PostgreSQL Operator Helm chart for Kubernetes
+type: application
+version: 0.1.0
+appVersion: 4.3.2
+home: https://github.com/CrunchyData/postgres-operator
+icon: https://github.com/CrunchyData/postgres-operator/raw/master/crunchy_logo.png
+keywords:
+    - PostgreSQL
```
Comment (jkatz) [inline installers/helm/Chart.yaml:10]: Also add:

Postgres
SQL
NoSQL
RDBMS

File: installers/helm/README.md (hunk 3 above)
Comment (jkatz) [reply installers/helm/README.md:12]: Helm v3

File: installers/helm/README.md [hunk 8]
```diff
@@ -0,0 +1,58 @@
+# Crunchy PostgreSQL Operator
+
+This Helm chart uses the `pgo-deployer` container to install the Crunchy PostgreSQL
```
Comment (jkatz) [inline installers/helm/README.md:3]: This Helm chart installs the Crunchy PostgreSQL Operator by using its “pgo-deployer” container.

File: installers/helm/templates/NOTES.txt (hunk 5 above)
Comment (jkatz) [reply installers/helm/templates/NOTES.txt:21]: Did you create it?

I don’t think we currently have ascii art of the Crunchy hippo

File: installers/helm/templates/postgres-operator-install.yaml (hunk 6 above)
Comment (jkatz) [reply installers/helm/templates/postgres-operator-install.yaml:12]: Yes, users should know what happens in the success and failure case.

File: installers/helm/templates/NOTES.txt (hunk 5 above)
Comment (jmckulk) [reply installers/helm/templates/NOTES.txt:21]: @jkatz no - i think Andrew found it somewhere on the webs

File: docs/content/installation/other/helm.md [hunk 9]
```diff
@@ -0,0 +1,234 @@
...
+{{% notice tip %}}
+You can also view the logs as the job is running by using the `kubectl -f`
+follow flag:
+```shell
+kubectl logs -n <namespace> job.batch/pgo-deploy -f
+```
+{{% /notice %}}
+
+
+These logs will provided feedback if there are any misconfigurations in your
```
Comment (andrewlecuyer) [inline docs/content/installation/other/helm.md:230]: typo: `provided` -> `provide`

File: installers/helm/README.md [hunk 10]
```diff
@@ -0,0 +1,58 @@
...
+## Configuraiton 
+
+The following shows the configurable parameters that are relevant to the Helm
+Chart. A full list of all Crunchy PostgreSQL Operator configuration options can
+be found in the [documentation](https://crunchydata.github.io/postgres-operator/latest/installation/configuration/).
+
+| Name | Default | Description |
+| ---- | ------- | ----------- |
+| fullnameOverride | "" |  |
+| rbac.create | true | If false rbac will not be created. RBAC resources will need to be created manually and bound to `serviceAccount.name` |
```
Comment (andrewlecuyer) [inline installers/helm/README.md:51]: Inconsistent capitalization of `RBAC` here.

File: installers/helm/README.md [hunk 11]
```diff
@@ -0,0 +1,58 @@
...
+| rbac.useClusterAdmin | false | If enabled the ServiceAccount will be given cluster-admin privileges. |
+| serviceAccount.create | true | If false a service account will not be created. ServiceAccount will need to be created manually |
+| serviceAccount.name | "" | Use to override the default service account name. If serviceAccount.create is false this service account will be used. |
```
Comment (andrewlecuyer) [inline installers/helm/README.md:54]: We mention `ServiceAccount` in some places, but then `service account` in others.  Can we make this consistent?

File: installers/helm/values.yaml [hunk 12]
```diff
@@ -0,0 +1,148 @@
+---
+# ======================
+# Installer Controls
+# ======================
+fullnameOverride: ""
+
+rbac:
+    # rbac.create: if false rbac resources should be in place
+    create: true
+    # rbac.useClusterAdmin: creates a ClusterRoleBinding giving cluster-admin to the ServiceAccount
```
Comment (andrewlecuyer) [inline installers/helm/values.yaml:10]: Might want to specify the specific ServiceAccount here for clarity.

File: docs/content/installation/other/helm.md [hunk 13]
```diff
@@ -0,0 +1,234 @@
...
+{{% notice tip %}}
+Take note of the `name` used when installing, this `name` will be used to
+upgrade and uninstall the PostgreSQL Operator.
+{{% /notice %}}
+
+### Install the [`pgo` Client]({{< relref "/installation/pgo-client" >}})
+
+To use the [`pgo` Client]({{< relref "/installation/pgo-client" >}}),
+there are a few additional steps to take in order to get it to work with you
+PostgreSQL Operator installation. For convenience, you can download and run the
```
Comment (cbandy) [inline docs/content/installation/other/helm.md:92]:
```suggestion
there are a few additional steps to take in order to get it to work with your
PostgreSQL Operator installation. For convenience, you can download and run the
```

File: docs/content/installation/other/helm.md [hunk 14]
```diff
@@ -0,0 +1,234 @@
...
+
+```shell
+export PGOUSER=$HOME/.pgo/$PGO_OPERATOR_NAMESPACE/pgouser
+export PGO_CA_CERT=$HOME/.pgo/$PGO_OPERATOR_NAMESPACE/client.crt
+export PGO_CLIENT_CERT=$HOME/.pgo/$PGO_OPERATOR_NAMESPACE/client.crt
+export PGO_CLIENT_KEY=$HOME/.pgo/$PGO_OPERATOR_NAMESPACE/client.key
+```
+
+For convenience, after the script has finished, you can permanently at these
+environmental variables to your environment:
```
Comment (cbandy) [inline docs/content/installation/other/helm.md:134]:
```suggestion
For convenience, after the script has finished, you can permanently add these
environmental variables to your environment:
```

File: docs/content/installation/other/helm.md [hunk 15]
```diff
@@ -0,0 +1,234 @@
...
+EOF
+```
+
+By default, the `client-setup.sh` script targets the user that is stored in the
+`pgouser-admin` secret in the `pgo` (`$PGO_OPERATOR_NAMESPACE`) Namespace. If
+you wish to use a different Secret, you can set the `PGO_USER_ADMIN`
+environmental variable.
+
+For more detailed information about [installing the `pgo` client]({{< relref "/installation/pgo-client" >}}),
+please see [Installing the `pgo` client]({{< relref "/installation/pgo-client" >}}).
```
Comment (cbandy) [inline docs/content/installation/other/helm.md:153]: This seems redundant. Perhaps the following?

```suggestion
For more detailed information,
please see [Installing the `pgo` client]({{< relref "/installation/pgo-client" >}}).
```

File: installers/helm/README.md [hunk 16]
```diff
@@ -0,0 +1,58 @@
...
+```
+cd postgres-operator/installers/helm
+helm uninstall postgres-operator . -n pgo
+```
+
+## Configuraiton 
+
+The following shows the configurable parameters that are relevant to the Helm
+Chart. A full list of all Crunchy PostgreSQL Operator configuration options can
+be found in the [documentation](https://crunchydata.github.io/postgres-operator/latest/installation/configuration/).
```
Comment (cbandy) [inline installers/helm/README.md:46]: Should this link to somewhere in https://access.crunchydata.com/documentation/ ?

File: installers/helm/templates/NOTES.txt [hunk 17]
```diff
@@ -0,0 +1,34 @@
...
+    #############%     %%%%%                      %%%%%%%%%%%
+      ###########%       %%%%%%%%%%%            %%%%%%%%%
+        #########%%     %%            %%%%%%%%%%%%%%%#
+          ########%%   %%                  %%%%%%%%%
+              ######%% %%                      %%%%%%
+                ####%%%                        %%%%%  %
+                      %%                         %%%%
+
+More information on using the postgres-operator can be found in the docs:
+https://crunchydata.github.io/postgres-operator/latest/
```
Comment (cbandy) [inline installers/helm/templates/NOTES.txt:34]: Same question about this URL.

File: installers/helm/README.md (hunk 16 above)
Comment (jkatz) [reply installers/helm/README.md:46]: Yes.

Comment (jkatz) [inline installers/helm/README.md:46]:
```suggestion
be found in the [documentation](https://access.crunchydata.com/documentation/postgres-operator/latest/installation/configuration/).
```

File: installers/helm/templates/NOTES.txt (hunk 17 above)
Comment (jkatz) [inline installers/helm/templates/NOTES.txt:34]:
```suggestion
https://access.crunchydata.com/documentation/postgres-operator/
```

**Pull request merged**